mods/carts/cart_entity.lua

```diff
@@ -193,7 +193,6 @@ local function rail_on_step(self, dtime)
 		local flo_old = vector.round(self.old_pos)
 		if vector.equals(flo_pos, flo_old) then
 			-- Do not check one node multiple times
-			rail_on_step_event(railparams.on_step, self, dtime)
```

Comment:
This case is not gerantueed to be called everytime. It would be required to integrate this into the pathfinder code to make sure each node's on_step is at least called once. I'm not willing to do it in this pull due to noticeably longer execution time.